crates/starknet-os/src/starknet/business_logic/fact_state/state.rs

```diff
@@ -430,22 +425,20 @@ where
         Err(StateError::UndeclaredClassHash(ClassHash(compiled_class_hash.0)))
     }
 
-    async fn get_storage_at_async(
-        &mut self,
-        contract_address: ContractAddress,
-        key: StorageKey,
-    ) -> StateResult<StarkFelt> {
-        let storage_key: TreeIndex = felt_api2vm(*key.0.key()).to_biguint();
+    async fn get_storage_at_async(&self, contract_address: ContractAddress, key: StorageKey) -> StateResult<Felt252> {
+        let storage_key: TreeIndex = key.0.key().to_biguint();
+
+        let mut ffc = self.ffc.clone();
```

Comment:
This is because the trait now forces this method to take `&self` instead of `&mut self`. The FFC is built around an `Arc<Mutex<Storage>>` so we can just deal with this problem by cloning the FFC and then modifying the clone.